tests/test_download.py

```diff
@@ -73,15 +71,11 @@ def setUp(self):
     self.target_data_length = len(self.target_data)
 
     # Launch a SimpleHTTPServer (serves files in the current dir).
-    self.PORT = random.randint(30000, 45000)
-    self.server_proc = popen_python(['simple_server.py', str(self.PORT)])
-    logger.info('\n\tServer process started.')
-    logger.info('\tServer process id: ' + str(self.server_proc.pid))
-    logger.info('\tServing on port: ' + str(self.PORT))
-    junk, rel_target_filepath = os.path.split(target_filepath)
-    self.url = 'http://localhost:'+str(self.PORT)+'/'+rel_target_filepath
+    self.server_process_handler = utils.TestServerProcess(log=logger)
 
-    utils.wait_for_server('localhost', self.PORT)
+    rel_target_filepath = os.path.basename(target_filepath)
+    self.url = 'http://localhost:' \
+        + str(self.server_process_handler.port) + '/' + rel_target_filepath
```

Comment:
Nice. 

I'd guess we don't even need to store the url during setup in many tests now that we can get the port number from the `server_process_handler`? 👍

Reply:
Well, 3 out of 4 tests in `test_download.py` are using the `self.url.` 
That's why it make sense to keep it as a variable.